source/common/conn_pool/conn_pool_base.h

```diff
@@ -80,6 +82,9 @@ class ActiveClient : public LinkedObject<ActiveClient>,
   // Returns the number of active streams on this connection.
   virtual uint32_t numActiveStreams() const PURE;
 
+  // Request tracking methods
+  virtual void trackRequest();
```

Comment:
This code is at the wrong level; this should be in `source/common/http/conn_pool_base.h, in the `ActiveClient` declared there.